### PR TITLE
feat: implement initial support for SSR

### PR DESCRIPTION
**Pull request by kyubisation**
This is the initial implementation for SSR testing.

### CODE REVIEW

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Thanks for starting this! I think some leftovers need to be handled.

Imports in [src/components/core/testing/a11y-tree-snapshot.ts](https://github.com/lyne-design-system/lyne-components/pull/2437/files#diff-016aa2786eca477f091d22b23ab3a99fded63a6c497b34fb2b4315f8b37b5a76) have to be fixed

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Thanks for starting this! I think some leftovers need to be handled.

Imports in [src/components/core/testing/a11y-tree-snapshot.ts](https://github.com/lyne-design-system/lyne-components/pull/2437/files#diff-016aa2786eca477f091d22b23ab3a99fded63a6c497b34fb2b4315f8b37b5a76) have to be fixed

File: src/components/accordion/accordion.e2e.ts [hunk 1]
```diff
@@ -47,30 +52,30 @@ describe('sbb-accordion', () => {
     let panels: SbbExpansionPanelElement[];
 
     element.querySelector('sbb-expansion-panel')!.remove();
-    await waitForLitRender(element);
+    await element.updateComplete;
```
[the remark this answers is not in the document]
Comment (kyubisation) [reply src/components/accordion/accordion.e2e.ts:55]: I wanted to check, whether that would be enough. I will revert it.

File: web-test-runner.config.js [hunk 2]
```diff
@@ -24,12 +25,34 @@ const browsers = isCIEnvironment
...
+</html>
+`;
+
 /** @type {import('@web/test-runner').TestRunnerConfig} */
 export default {
   files: ['src/**/*.{e2e,spec}.ts'],
   groups: [
-    { name: 'spec', files: 'src/**/*.spec.ts' },
-    { name: 'e2e', files: 'src/**/*.e2e.ts' },
+    //{ name: 'spec', files: 'src/**/*.spec.ts' },
```
[the remark this answers is not in the document]
Comment (kyubisation) [reply web-test-runner.config.js:52]: I want to discuss this during refinement, as it has consequences.

File: web-test-runner.config.js [hunk 3]
```diff
@@ -20,46 +24,66 @@ const browsers = isCIEnvironment
...
+const groupNameOverride = process.argv.includes('--ssr-hydrated')
+  ? 'e2e-ssr-hydrated'
+  : process.argv.includes('--ssr-non-hydrated')
+    ? 'e2e-ssr-non-hydrated'
+    : null;
+
+const testRunnerHtml = (testFramework, _config, group) => `
+<html>
+  <head>
+    <meta name="testEnvironment" ${isDebugMode ? 'debug' : ''}>
```
Comment (jeripeierSBB) [inline web-test-runner.config.js:49]: Should html data- attributes be used?

File: src/components/icon/icon-base.ts [hunk 4]
```diff
@@ -26,23 +26,27 @@ export abstract class SbbIconBase extends UpdateScheduler(LitElement) {
    */
   @property({ attribute: 'no-sanitize', type: Boolean }) public noSanitize = false;
 
+  protected override createRenderRoot(): HTMLElement | DocumentFragment {
+    this.setAttribute('data-namespace', this._svgNamespace);
+    if (!this.hasAttribute('data-empty')) {
```
Comment (jeripeierSBB) [inline src/components/icon/icon-base.ts:31]: Fix this

File: web-test-runner.config.js [hunk 5]
```diff
@@ -20,46 +24,66 @@ const browsers = isCIEnvironment
...
   plugins: [vitePlugin(), a11ySnapshotPlugin()],
   testFramework: {
     config: {
-      timeout: '3000',
+      timeout: '6000',
       slow: '1000',
       failZero: true,
     },
   },
-  coverageConfig: {
```
Comment (jeripeierSBB) [inline web-test-runner.config.js:45]: Fix Coverage

File: web-test-runner.config.js (hunk 3 above)
Comment (kyubisation) [reply web-test-runner.config.js:49]: Will be considered in a separate PR.

**Pull request merged**